RxJavaTemplate/app/src/main/java/co/nimblehq/rxjava/storage/BaseSharedPreferencesImpl.kt

```diff
@@ -0,0 +1,39 @@
+package co.nimblehq.rxjava.storage
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import co.nimblehq.rxjava.extension.execute
+import co.nimblehq.rxjava.extension.put
+import javax.inject.Inject
+
+// TODO: For demo purpose only, replace with actual value that needs to be stored securely
+private const val PREF_TEST_ID = "PREF_TEST_ID"
+
+class BaseSharedPreferencesImpl @Inject constructor(applicationContext: Context) :
```

Comment:
I think we don't need to create `interface` and `Impl` class. Why don't we create abstract class? 🤔 

One more thing, we should pass the `prefsName` from the constructor.